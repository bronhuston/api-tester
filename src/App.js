import React, {Component} from 'react';
import logo from './logo.svg';
import './App.css';

import UrlInput from './UrlInput';
import RequestBodyInput from './RequestBodyInput';

class App extends Component {
  constructor() {
    super();
    this.state = {
      url: '',
      requestBodyInput: ''
    };
  }

  handleChange = (key, e) => {
    console.log(key, e.target.value);
    this.setState({[key]: e.target.value});
  };

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo"/>
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          <UrlInput url={this.state.url} onChange={(e) => this.handleChange('url', e)}/>
        </p>
        <div>
          <RequestBodyInput requestBody={this.state.requestBody} onChange={(e) => this.handleChange('requestBodyInput', e)}/>
        </div>
      </div>
    );
  }
}

export default App;
