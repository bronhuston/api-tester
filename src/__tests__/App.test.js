import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders snapshots with shallow enzyme', () => {
  expect(shallow(<App />)).toMatchSnapshot();
});

it('renders with react-test-renderer', () => {
  expect(renderer.create(<App />).toJSON()).toMatchSnapshot();
});