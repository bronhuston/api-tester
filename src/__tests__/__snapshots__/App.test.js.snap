// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshots with shallow enzyme 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App"
>
    <div
        className="App-header"
    >
        <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
        />
        <h2>
            Welcome to React
        </h2>
    </div>
    <p
        className="App-intro"
    >
        <UrlInput
            onChange={[Function]}
            url=""
        />
    </p>
    <div>
        <RequestBodyInput
            onChange={[Function]}
            requestBody={undefined}
        />
    </div>
    <div>
        <button
            onClick={[Function]}
        >
            Generate Things
        </button>
    </div>
    <div
        style={
            Object {
                "border": 1,
                "padding": 5,
              }
        }
    >
        
    </div>
</div>,
  "nodes": Array [
    <div
      className="App"
>
      <div
            className="App-header"
      >
            <img
                  alt="logo"
                  className="App-logo"
                  src="logo.svg"
            />
            <h2>
                  Welcome to React
            </h2>
      </div>
      <p
            className="App-intro"
      >
            <UrlInput
                  onChange={[Function]}
                  url=""
            />
      </p>
      <div>
            <RequestBodyInput
                  onChange={[Function]}
                  requestBody={undefined}
            />
      </div>
      <div>
            <button
                  onClick={[Function]}
            >
                  Generate Things
            </button>
      </div>
      <div
            style={
                  Object {
                        "border": 1,
                        "padding": 5,
                      }
            }
      >
            
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 17,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "generateThings": [Function],
        "handleChange": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "requestBodyInput": "",
          "url": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="App"
>
          <div
                    className="App-header"
          >
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg"
                    />
                    <h2>
                              Welcome to React
                    </h2>
          </div>
          <p
                    className="App-intro"
          >
                    <UrlInput
                              onChange={[Function]}
                              url=""
                    />
          </p>
          <div>
                    <RequestBodyInput
                              onChange={[Function]}
                              requestBody={undefined}
                    />
          </div>
          <div>
                    <button
                              onClick={[Function]}
                    >
                              Generate Things
                    </button>
          </div>
          <div
                    style={
                              Object {
                                        "border": 1,
                                        "padding": 5,
                                      }
                    }
          >
                    
          </div>
</div>,
        "_debugID": 18,
        "_renderedOutput": <div
          className="App"
>
          <div
                    className="App-header"
          >
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg"
                    />
                    <h2>
                              Welcome to React
                    </h2>
          </div>
          <p
                    className="App-intro"
          >
                    <UrlInput
                              onChange={[Function]}
                              url=""
                    />
          </p>
          <div>
                    <RequestBodyInput
                              onChange={[Function]}
                              requestBody={undefined}
                    />
          </div>
          <div>
                    <button
                              onClick={[Function]}
                    >
                              Generate Things
                    </button>
          </div>
          <div
                    style={
                              Object {
                                        "border": 1,
                                        "padding": 5,
                                      }
                    }
          >
                    
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;

exports[`renders with react-test-renderer 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Welcome to React
    </h2>
  </div>
  <p
    className="App-intro"
  >
    <label>
      URL
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
  </p>
  <div>
    <textarea
      className="App-request-body-input"
      onChange={[Function]}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Generate Things
    </button>
  </div>
  <div
    style={
      Object {
        "border": 1,
        "padding": 5,
      }
    }
  />
</div>
`;
